src/databases/functions/users/user-block-by-admin.js

```diff
@@ -0,0 +1,12 @@
+const {User} = require('../../models');
+
+module.exports = async (userId) => {
+  if (!userId) return null;
+  try {
+    const result = await User.update({blocked_by_admin: true}, {where: {user_id: userId}});
+    return result;
+  } catch (error) {
+    console.error(`Failed to block user with id ${userId}: `, error);
+    return null;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `userBlockByAdmin` is well written and handles the case where `userId` might be null. However, it's not a good practice to return `null` in case of an error. Instead, you should throw an error so that the caller can handle it appropriately. Also, logging errors directly to the console may not be ideal for production environments. Consider using a dedicated logging library or service that provides more control over log levels, formats, and destinations.

```diff
-    console.error(`Failed to block user with id ${userId}: `, error);
-    return null;
+    throw new Error(`Failed to block user with id ${userId}: ${error}`);
```
Also, consider returning the updated user object instead of the result of the update operation. This would provide more information to the caller about the state of the user after the operation.

```diff
-    const result = await User.update({blocked_by_admin: true}, {where: {user_id: userId}});
-    return result;
+    await User.update({blocked_by_admin: true}, {where: {user_id: userId}});
+    return User.findOne({where: {user_id: userId}});
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->